ods_ci/tests/Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot

```diff
@@ -135,21 +135,24 @@ Run Cell And Get Output
     RETURN    ${output}
 
 Python Version Check
-  [Arguments]  ${expected_version}=3.8
-  Add And Run JupyterLab Code Cell In Active Notebook  !python --version
-  Wait Until JupyterLab Code Cell Is Not Active
-  #Get the text of the last output cell
-  ${output} =  Get Text  (//div[contains(@class,"jp-OutputArea-output")])[last()]
-  #start is inclusive, end exclusive, get x.y from Python x.y.z string
-  ${output} =  Fetch From Right  ${output}  ${SPACE}
-  ${vers} =  Get Substring  ${output}  0  3
-  ${status} =  Run Keyword And Return Status  Should Match  ${vers}  ${expected_version}
-  IF  '${status}' == 'FAIL'  Run Keyword And Continue On Failure  FAIL  "Expected Python at version ${expected_version}, but found at v ${vers}"
-
+    [Documentation]    Checks that the X.Y python version of the current Jupyterlab instance matches an expected one
+    [Arguments]    ${expected_version}=3.8
+    ${vers} =  Get XY Python Version From Jupyterlab
+    ${status} =  Run Keyword And Return Status  Should Match  ${vers}  ${expected_version}
+    IF  '${status}' == 'FAIL'  Run Keyword And Continue On Failure  FAIL  "Expected Python at version ${expected_version}, but found at v ${vers}"    # robocop: disable
+
+Get XY Python Version From Jupyterlab
+    [Documentation]    Fetches the X.Y Python version from the current Jupyterlab instance
+    ${output}=    Run Cell And Get Output    !python --version
```

Comment:
I'd do this

```
% python -c 'import sys; print(*sys.version_info[:2], sep=".")'
3.11
```

to avoid robot string manipulation at all costs. But it's been already coded differently, so I agree leave it be.

Reply:
yes, I wanted to fetch the string directly but a bunch of other tests rely on this keyword and I didn't want to change its behaviour too much in fear of breaking something. I think we can leave it be for now, RF's string manipulation is Python string manipulation at the end of the day

Reply:
$$\displaystyle {\hat {H}}\psi =E\psi$$

(everything that was, is, will be, and can be, is contained in the expression above, no need to write anything more)